### ISSUE TITLE
Deprecated Android embedding version

### ISSUE THREAD

**Issue by ikurek** [ably/ably-flutter#311]
Project uses Flutter Android Embedding V1, which has been deprecated for a while, and with latest Flutter versions (after 2.8.1) an attempt to run example application will fail with this message:

```
This app is using a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to migrate this app to the V2 embedding.
Take a look at the docs for migrating an app: https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects
The plugin `device_info_plus` requires your app to be migrated to the Android embedding v2. Follow the steps on https://flutter.dev/go/android-project-migration and re-run this command.
```

According to Flutter team a migration should be made to Android Embedding V2. As a temporary workaround, the example can be run by downgrading to older versions of Flutter engine

**Comment by QuintinWillison** [comment]
@ikurek I would perhaps go as far as to label this as a `bug`... for this repository, as a whole, the example apps are pretty essential. What do you think?

**Comment by ikurek** [comment]
@QuintinWillison Seems reasonable, since this is actually breaking for end users. I'm already working on a fix.

**Comment by QuintinWillison** [comment]
Fixed by #312 as far as I can tell, @ikurek. It seems this issue wasn't linked to the PR.
